source/common/http/filter_manager.cc

```diff
@@ -241,6 +242,25 @@ void ActiveStreamFilterBase::clearRouteCache() {
   parent_.filter_manager_callbacks_.clearRouteCache();
 }
 
+void ActiveStreamFilterBase::evaluateMatchTreeWithNewData(
+    std::function<void(HttpMatchingDataImpl&)> update_func) {
+  if (match_tree_evaluated_ || !matching_data_) {
+    return;
+  }
+
+  update_func(*matching_data_);
+
+  const auto match_result = Matcher::evaluateMatch<HttpMatchingData>(*match_tree_, *matching_data_);
+
+  match_tree_evaluated_ = match_result.match_state_ == Matcher::MatchState::MatchComplete;
+
+  if (match_tree_evaluated_ && match_result.result_) {
+    if (dynamic_cast<SkipAction*>(match_result.result_.get())) {
```

Comment:
Generally not thrilled with implicit RTTI usage like this. I thought we discussed potentially exposing the type URL as an interface method and then allowing branching on that when necessary?

Reply:
Yeah happy to switch to the type URL approach instead. I think initially I shied away from using the type URL to avoid long string comparisons, but I guess I'm not sure how that compares to a dynamic_cast. Just so I understand, what is the motivation behind avoiding RTII?

Reply:
Mostly that it's somewhat magical but generally a style preference. If you want to do at this way you should use `typeid` as it's more clear what you are doing.

Reply:
I'll give the string based approach a go for now, if it becomes a perf issue we can revisit